Source/Evolve/WarpXEvolve.cpp

```diff
@@ -955,3 +959,47 @@ WarpX::VayDeposition ()
     "WarpX::CurrentCorrection: requires WarpX build with spectral solver support.");
 #endif
 }
+
+void
+WarpX::CheckSignals()
+{
+    // We assume that signals will definitely be delivered to rank 0,
+    // and may be delivered to other ranks as well. For coordination,
+    // we process them according to when they're received by rank 0.
+    if (amrex::ParallelDescriptor::MyProc() == 0)
+    {
+        for (int i = 0; i < 32; ++i)
+        {
+            // Read into a local temporary to ensure the same value is
+            // used throughout. Atomically exchange it with zero to
+            // unset the flag without risking loss of a signal - if a
+            // signal arrives after this, it will be handled the next
+            // time this function is called.
+            bool signal_i_received = signal_received_flags[i].exchange(false);
+
+            if (signal_i_received)
+            {
+                signal_requested_break |= signal_conf_requests_break[i];
+                signal_requested_checkpoint |= signal_conf_requests_checkpoint[i];
+            }
+        }
+    }
+
+    auto comm = amrex::ParallelDescriptor::Communicator();
+
+    MPI_Ibcast(&signal_requested_break     , 1, MPI_CXX_BOOL, 0, comm, &signal_mpi_ibcast_requests[0]);
+    MPI_Ibcast(&signal_requested_checkpoint, 1, MPI_CXX_BOOL, 0, comm, &signal_mpi_ibcast_requests[1]);
```

Comment:
There's currently no `amrex::ParallelDescriptor::Abcast` for this to call. Will it be acceptable to go ahead with this here, and follow on in separate PRs to add that to AMReX and update these calls?

Reply:
Discussed today: slightly more preferred by us to make this a single Ibcast.
Let's use enums to have named/readable array elements.